crates/ty_python_semantic/src/types/builder.rs

```diff
@@ -204,19 +204,41 @@ enum ReduceResult<'db> {
 // representations, so that we can make large unions of literals fast in all operations.
 const MAX_UNION_LITERALS: usize = 200;
 
+pub(crate) enum UnionStrategy {
```

Comment:
I think it would be beneficial to have documentation outlining the different strategies and their respective needs and when to use which one